Pellicola/Classes/ImagesDataStorage.swift

```diff
@@ -38,28 +40,54 @@ class ImagesDataStorage: NSObject {
         
         if let limit = limit, images.count >= limit { return }
         
-        images[identifier] = OrderedImage(image: image, index: index)
+        let orderedImage = OrderedImage(index: index)
+        images[identifier] = orderedImage
         index += 1
+        
+        let saveItem = DispatchWorkItem { [weak self] in
```

Comment:
With the new implementation all methods from this class can now fail. These failures are not reported/handled in any way. Is this OK?